refactor: OPTIC-927: remove FF_DEV_2432 feature flag

Gondragos opened the pull request:
### Description

This pull request removes the `ff_front_dev_2432_auto_save_polygon_draft_210622_short` feature flag. All associated conditional logic and tests dependent on this feature flag have also been cleaned up.

- Improved the undo functionality in regression tests for image draw to support multiple draw points.  
- Ensured proper handling of the 'closed' property for polygonal shapes in e2e test cases. This guarantees accurate test execution and results with explicitly closed polygon shapes.  
- Fixed the problem where appears unnecessary labels unselecting [[1]](https://github.com/HumanSignal/label-studio/pull/7229/files#diff-c954aa71b99a7d8e33373aadf2ef578a22e2a3d2723c610bfc859806c1899b89R665-R666)

| **Key** | `ff_front_dev_2432_auto_save_polygon_draft_210622_short` |
|---|---|
| **Variable Name** | `FF_DEV_2432` |
| **Ends up as** | 🟩 `True` |
| **Description** | Allowing saving not closed polygons and managing the process of re-initialization drawing mode. |

### Reason for change
The feature flag is no longer in use, and keeping it creates unnecessary code complexity. Removing it simplifies the codebase.

### Rollout strategy
As the feature flag is no longer active, this change does not require any specific rollout plan.

### Testing:  
- Automated regression and end-to-end tests were updated to reflect the changes.  
- Verified that all tests pass successfully with the modified undo logic and updated test cases for closed polygons.  

### Risks
There are minimal risks as the feature flag has been inactive. However, ensure a thorough review of the deletions to confirm no unintended functionality changes are introduced.

### Reviewer notes
Focus on ensuring proper removal of all related code and tests. Verify that there are no lingering references to the deleted feature flag.

### General notes
This is a cleanup task with no visible changes to the application's functionality. It contributes towards maintaining a tidy and efficient codebase

<!--

This description MUST be filled out for a PR to receive a review. Its primary purposes are:

 - to enable your reviewer to review your code easily, and
 - to convince your reviewer that your code works as intended.

Some pointers to think about when filling out your PR description:
 - Reason for change: Description of problem and solution
 - Screenshots: All visible changes should include screenshots.
 - Rollout strategy: How will this code be rolled out? Feature flags / env var / other
 - Testing: Description of how this is being verified
 - Risks: Are there any known risks associated with this change, eg to security or performance?
 - Reviewer notes: Any info to help reviewers approve the PR
 - General notes: Any info to help onlookers understand the code, or callouts to significant portions.

You may use AI tools such as Copilot Actions to assist with writing your PR description (see https://docs.github.com/en/copilot/using-github-copilot/using-github-copilot-for-pull-requests/creating-a-pull-request-summary-with-github-copilot); however, an AI summary isn't enough by itself. You'll need to provide your reviewer with strong evidence that your code works as intended, which requires actually running the code and showing that it works.

-->

Gondragos committed:
refactor: remove FF_DEV_2432 feature flag

Removed the `ff_front_dev_2432_auto_save_polygon_draft_210622_short` feature flag and all associated conditional logic and tests.

Gondragos committed:
fix: add missing 'closed' property to polygons

The 'closed' property was added to polygon configurations in e2e tests to ensure proper functionality during testing. This change fixes any potential issues arising from missing closure definitions in polygon data for the examples.

Gondragos committed:
fix: handle unfinished regions in selection update

Ensure that unfinished regions take precedence during selection updates. This change adds a condition to check for drawing regions before updating the results, preventing potential inconsistencies in region selection behavior.

Gondragos committed:
fix: ensure polygon shapes are closed in tests

Updated e2e tests for image gestures to explicitly include the `closed` property in polygon shape expectations.

Gondragos committed:
fix: ensure 'closed' property is set in test cases

Added the missing 'closed' property in multiple e2e test files to ensure polyline shapes are properly identified as closed.

Gondragos committed:
fix: improve undo functionality for image draw tests

Enhanced undo logic in the regression test to better handle multiple points by iterating undo steps based on the number of drawn points.

Gondragos commented:
/git merge 
> [Workflow run](https://github.com/HumanSignal/label-studio/actions/runs/13902076153)
> Successfully merged:  3 files changed, 31 insertions(+), 4 deletions(-)

Gondragos committed:
Merge branch 'develop' into 'fb-OPTIC-927/stale-ff'

Workflow run: https://github.com/HumanSignal/label-studio/actions/runs/13902076153

Gondragos committed:
fix: remove stale feature flag for auto-save polygon

Removed the stale feature flag `ff_front_dev_2432_auto_save_polygon_draft_210622_short` as it is no longer in use.

Gondragos commented:
/git merge
> [Workflow run](https://github.com/HumanSignal/label-studio/actions/runs/13981716205)
> Successfully merged:  create mode 100644 web/tools/design-tokens-converter/package.json

Gondragos committed:
Merge branch 'develop' into 'fb-OPTIC-927/stale-ff'

Workflow run: https://github.com/HumanSignal/label-studio/actions/runs/13981716205

borisheartex commented:
/fm sync
> [Workflow run](https://github.com/HumanSignal/label-studio/actions/runs/13986676955)

robot-ci-heartex committed:
Sync Follow Merge dependencies

Workflow run: https://github.com/HumanSignal/label-studio/actions/runs/13986676955

Gondragos committed:
fix: correct region resizing test and improve logic

Gondragos committed:
Merge remote-tracking branch 'origin/fb-OPTIC-927/stale-ff' into fb-OPTIC-927/stale-ff

Gondragos committed:
fix: correct method name in e2e test assertions

Replaced `dontElement` with `dontSeeElement` in the image transformer test to fix a typo in the assertion method. Additionally, added the `dontSeeElement` method to `AtSidebar.js` to ensure consistent functionality across test fragments.

Gondragos committed:
fix: ensure incomplete state badge is hidden

Updated the image transformer test to verify that the incomplete state badge does not appear in the sidebar after drawing a complete shape. This change improves the reliability of the test by explicitly checking the absence of the badge.

Gondragos committed:
test: add locator for incomplete state badge

Added a new locator `_incompleteStateBadge` to handle elements indicating an incomplete state. This improves test coverage for scenarios involving incomplete nodes in the sidebar.

Gondragos commented on web/libs/editor/src/stores/Annotation/Annotation.js:
```diff
@@ -659,12 +658,11 @@ const _Annotation = types
-        const areas = Array.from(self.areas.values());
-        const filtered = areas.filter((area) => area.isDrawing);
+      const areas = Array.from(self.areas.values());
+      const filtered = areas.filter((area) => area.isDrawing);
```
```suggestion
      // It should find just one unfinished region, but just in case we work with array
      const filtered = areas.filter((area) => area.isDrawing);
```

Gondragos commented on web/libs/editor/src/stores/Annotation/Annotation.js:
```diff
@@ -659,12 +658,11 @@ const _Annotation = types
 
-        self.regionStore.selection._updateResultsFromRegions(filtered);
-      }
+      // An unfinished region takes control if it exists.
```
```suggestion
      // Update UI to reflect the state of an unfinished region in case if it exists
```

Commit:
Update comments

Gondragos commented:
/git merge
> [Workflow run](https://github.com/HumanSignal/label-studio/actions/runs/14076305263)
> Successfully merged:  create mode 100644 web/libs/ui/src/lib/code-block/code-block.tsx

Gondragos committed:
Merge branch 'develop' into 'fb-OPTIC-927/stale-ff'

Workflow run: https://github.com/HumanSignal/label-studio/actions/runs/14076305263

borisheartex commented:
/fm sync
> [Workflow run](https://github.com/HumanSignal/label-studio/actions/runs/14077631564)

robot-ci-heartex committed:
Sync Follow Merge dependencies

Workflow run: https://github.com/HumanSignal/label-studio/actions/runs/14077631564

Gondragos commented:
/git merge
> [Workflow run](https://github.com/HumanSignal/label-studio/actions/runs/14086870059)
> Successfully merged:  create mode 100644 web/libs/ui/src/tokens/tokens.stories.tsx

Gondragos committed:
Merge branch 'develop' into 'fb-OPTIC-927/stale-ff'

Workflow run: https://github.com/HumanSignal/label-studio/actions/runs/14086870059
